dotnet-authserver/src/TeacherIdentity.AuthServer/Pages/Admin/MergeUser/Confirm.cshtml.cs

```diff
@@ -0,0 +1,127 @@
+using Microsoft.AspNetCore.Mvc;
+using Microsoft.AspNetCore.Mvc.Filters;
+using Microsoft.AspNetCore.Mvc.RazorPages;
+using Microsoft.EntityFrameworkCore;
+using TeacherIdentity.AuthServer.Models;
+
+namespace TeacherIdentity.AuthServer.Pages.Admin.MergeUser;
+
+public class Confirm : PageModel
+{
+    public static string MergeTrnKey = "MergeTrn";
+    public new User? User { get; set; }
+    public User? UserToMerge { get; set; }
+
+    private readonly TeacherIdentityServerDbContext _dbContext;
+    private readonly IClock _clock;
+
+    public Confirm(TeacherIdentityServerDbContext dbContext, IClock clock)
+    {
+        _dbContext = dbContext;
+        _clock = clock;
+    }
+
+    [FromRoute]
+    public Guid UserId { get; set; }
+
+    [FromRoute]
+    public Guid UserIdToMerge { get; set; }
+
+    public void OnGet()
+    {
+    }
+
+    public async Task<IActionResult> OnPost()
+    {
+        if (!ModelState.IsValid)
+        {
+            return this.PageWithErrors();
+        }
+
+        MergeUsers(User!, UserToMerge!);
+
+        await _dbContext.SaveChangesAsync();
+
+        HttpContext.Session.Remove(MergeTrnKey);
```

Comment:
We want to unset this value to prevent unwanted interaction with subsequent merges